controllers/hosting/model_reconciler.go

```diff
@@ -392,6 +392,11 @@ func (r *modelReconciler) getContainerDefinitions(model *commonv1.Model, contain
 // This also returns an error if the primary container definition cannot be found.
 func (r *modelReconciler) getPrimaryContainerDefinition(model *commonv1.Model, containers map[string]*commonv1.ContainerDefinition) (*commonv1.ContainerDefinition, error) {
 
+	// For multi-container, don't expect a primary container
+	if model.PrimaryContainer == nil && len(model.Containers) > 1 {
```

Comment:
Can we ignore this field if there are more than one containers rather than expecting `model.PrimaryContainer` to be nil? 

because there is no guarantee. from the old code, it looks like one needs to specify one container as primary when using multi-container

Reply:
Hmm yeah actually that's a good idea. I might just gate it from the outer method rather than modifying these helper methods